qutebrowser/keyinput/basekeyparser.py

```diff
@@ -366,6 +365,7 @@ def on_keyconfig_changed(self, mode):
 
     def clear_keystring(self):
         """Clear the currently entered key sequence."""
-        self._debug_log("discarding keystring '{}'.".format(self._keystring))
-        self._keystring = ''
-        self.keystring_updated.emit(self._keystring)
+        if self._keystring:
+            self._debug_log("discarding keystring '{}'.".format(self._keystring))
+            self._keystring = ''
+            self.keystring_updated.emit(self._keystring)
```

Comment:
I think this could cause `keystring_updated` to be emitted twice now in some situations, no? Once when it wasn't before in `_handle_single_key` or `_handle_ambiguous_match`, and then again in `handle` (which unconditionally emits it after calling `_handle_*`)